### PR TITLE
Feature/milestone 4

### PR DESCRIPTION
**Pull request by zilton7**
In This Milestone:
- Create a logic directory.
- Implement the 'calculate' module.
- Implement the 'operate' module.
- Add calculate to the App component.

Thanks for the review.

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable no-param-reassign, eqeqeq, no-undef */
```
Comment (bellom) [inline src/logic/calculate.js]: Inline comment: 
With the ESLint inline comment, you're are indirectly changing the linter's rules by disabling these rules, let's fix the codebase instead.

If you are using VSCode, you can delete the inline comment and hover over the error(configure eslint checks locally). And you will see your options. Click on the more info button that will take you to a GitHub repo where it is explained why you are getting this error. I find this feature very educational. Please, try to learn and implement the necessary rules so you don't have to keep this disabling line on top of each file. I believe you can do this! 💪😇

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable no-param-reassign, eqeqeq, no-undef, no-unused-vars */
```
Comment (bellom) [inline src/logic/operate.js]: With the ESLint inline comment, you're are indirectly changing the linter's rules by disabling these rules, let's fix the codebase instead.

File: src/logic/calculate.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import operate from './operate';
+
+const calculate = (buttonName, dataObj) => {
```
Comment (bellom) [inline src/logic/calculate.js:4]: According to the instructions on pathwright, you should make use of destructuring.
Example;  `{ total, next, operation } = dataObj `
image

File: src/logic/calculate.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  if (val.isNaN()) {
+    dataObj.total += val;
+  } else if (buttonName == '+/-') {
+    dataObj.total *= -1;
+    dataObj.next *= -1;
+  } else if (buttonName == 'AC') {
+    dataObj.total = '';
+    dataObj.next = '';
+    dataObj.operation = '';
+  } else if (buttonName == '=') {
```
Comment (bellom) [inline src/logic/calculate.js]: I think you should use `===`.
Example; `} else if (buttonName === '+/-') {`
we need to check the dataType and compares the two values.

File: src/logic/calculate.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    operate(total, next, operation);
+    dataObj.total = '';
+    dataObj.next = '';
+    dataObj.operation = '';
+  }
+  return dataObj;
```
Comment (bellom) [inline src/logic/calculate.js:63]: You need to add more conditions.
What if the buttonName is a digit or `.`.

Example: if the `buttonName`  = `.` we should return `0.`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @zilton7  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Invalid Submission 

### Hi Zil Norvilis :slightly_smiling_face: 

I did not review the changes you have worked on. This is because passing linter validation is hard requirement, for a pull request to be reviewed. Kindly refer to this link below:
https://microverse.zendesk.com/hc/en-us/articles/1500002085202-February-2nd-2021-linter-errors-as-a-reason-of-project-invalidation

#### Ensure you pass the linter validations, then submit another form for a code review.

### Happy coding :man_technologist: 

#### ``` Uchechukwu Henry Inyama ```

**Review by bellom — APPROVED** [review]
## Hi @zilton7  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**